internal/infrastructure/bot/handlers.go

```diff
@@ -271,14 +272,63 @@ func (b *Bot) PrivateSummary(i *discordgo.InteractionCreate) error {
 		return err
 	}
 
-	err = b.eventHandler.OnPrivateSummary(summary.PrivateSummaryRequest{
-		GuildID: gID,
-		UserID:  uID,
+	isSummaryWithoutRespawns := len(i.ApplicationCommandData().Options) < 1
+	if isSummaryWithoutRespawns {
+		err = b.eventHandler.OnPrivateSummary(summary.PrivateSummaryRequest{
+			GuildID: gID,
+			UserID:  uID,
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		spotNames := strings.Split(i.ApplicationCommandData().Options[0].StringValue(), ", ")
+		err = b.eventHandler.OnPrivateSummary(summary.PrivateSummaryRequest{
+			GuildID:   gID,
+			UserID:    uID,
+			SpotNames: spotNames,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = b.mgr.SessionForGuild(gID).FollowupMessageCreate(i.Interaction, false, &discordgo.WebhookParams{Content: "Check your DM!"})
+	return err
+
+}
+
+func (b *Bot) PrivateSummaryWithSpotNamesAutocomplete(i *discordgo.InteractionCreate) error {
+
+	userInput, index := collections.PoorMansFind(i.ApplicationCommandData().Options,
+		func(o *discordgo.ApplicationCommandInteractionDataOption) bool {
+			return o.Focused
+		})
+	if index == -1 {
+		return errors.New("none of the options were selected for autocompletion")
+	}
+
+	lastEntry := strings.Split(userInput.StringValue(), ",")[len(strings.Split(userInput.StringValue(), ","))-1]
```

Comment:
We're calling `strings.Split` twice here. While the compiler may optimize this for us, it's probably cleaner to do it once and then reuse the value:

```go
spotsInput := strings.Split(userInput.StringValue(), ",")
lastEntry := spotsInput[len(spotsInput)-1]
```